src/v/cluster/topic_table.cc

```diff
@@ -104,6 +104,12 @@ topic_table::apply(move_partition_replicas_cmd cmd, model::offset o) {
         return ss::make_ready_future<std::error_code>(errc::update_in_progress);
     }
 
+    // assignment is already up to date, this operation is NOP do not propagate
+    // delta
```

Comment:
can we update this comment? the comment makes it sound like an optimization has been added. but the commit message seems to imply some pretty subtle problems that can occur if this is ever removed.

I think that means we need to describe what the problem is in a little more detail than the commit message currently does. the most obvious question would be why isn't an empty/noop update handled the same as every other update such that it can be waited on to finish?

Reply:
I will update this comment and add a commit that will make this an optimization.